src/Umbraco.Web.BackOffice/Filters/OutgoingEditorModelEventAttribute.cs

```diff
@@ -4,47 +4,60 @@
 using Umbraco.Core;
 using Umbraco.Core.Security;
 using Umbraco.Web.Editors;
-using Umbraco.Web.Security;
 
 namespace Umbraco.Web.BackOffice.Filters
 {
     /// <summary>
     /// Used to emit outgoing editor model events
     /// </summary>
-    internal sealed class OutgoingEditorModelEventAttribute : ActionFilterAttribute
+    internal sealed class OutgoingEditorModelEventAttribute : TypeFilterAttribute
     {
-        private readonly IUmbracoContextAccessor _umbracoContextAccessor;
-        private readonly IBackOfficeSecurityAccessor _backofficeSecurityAccessor;
-
-        public OutgoingEditorModelEventAttribute(IUmbracoContextAccessor umbracoContextAccessor, IBackOfficeSecurityAccessor backofficeSecurityAccessor)
+        public OutgoingEditorModelEventAttribute() : base(typeof(OutgoingEditorModelEventFilter))
         {
-            _umbracoContextAccessor = umbracoContextAccessor ?? throw new ArgumentNullException(nameof(umbracoContextAccessor));
-            _backofficeSecurityAccessor = backofficeSecurityAccessor ?? throw new ArgumentNullException(nameof(backofficeSecurityAccessor));
         }
 
-        public override void OnActionExecuted(ActionExecutedContext context)
+
+        private class OutgoingEditorModelEventFilter : IResultFilter
```

Comment:
If you use `IActionFilter` here, you can still override `OnActionExecuted` and then all the changes to events can be reverted